### PR TITLE
Fix/avoid corner case

### PR DESCRIPTION
**Pull request by gsheasby**
**Goals (and why)**: 
- Avoid sweep getting stuck in an infinite loop (why? because infinite loops are bad)
- Actually run the test that would have caught this in circle again

**Implementation Description (bullets)**:
- Add memorySensitiveTest back to the least-stressed container
- Use RangeRequests.nextLexicographicName to avoid successive loop iterations containing the same row, which is a big problem if this is the only row remaining and it only contains tombstones, because we'll never get any farther.

**Concerns (what feedback would you like?)**:
- Do we need the extra tests? Are they at the right specificity?
- Do we need all the timeouts on the sweep tests? None of them take longer than 12 seconds (with batch size = false) on my IntelliJ.
- Is the release note messaging appropriate?

**Where should we start reviewing?**: GetCellTimestamps, which turns out to be the first file on the PR.

**Priority (whenever / two weeks / yesterday)**: ASAP

**The gory details (copied from the message of the commit that actually fixes the thing)
Fix an edge case that caused an infinite loop in
AbstractSweepTaskRunnerTest.testSweepLatestDeletedThrough.

If the range contains a row that has no columns (including if the row's
only column was deleted), the Thrift rowGetter query will return that
row, although the list of columns would be empty - the row's still
there, but it's tombstoned. Therefore, determineSafeRangeEndInclusive
would return that row's key as a safe range end (and not
Optional.Empty). So we run CqlExecutor.getTimestamps, get an empty
list, and iterate again through the loop.

Previously, we set rangeStart to the previous rangeEnd. Both ends of
the CqlExecutor.getTimestamps query are inclusive, so on the next
iteration we'd get the empty row again, and rangeEnd and rangeStart
would never change.

Using RangeRequests.nextLexicographicName avoids fetching the same
empty row again, so we can continue making forward progress.

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/2744)

### CODE REVIEW

File: atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/sweep/AbstractSweepTaskRunnerTest.java [hunk 1]
```diff
@@ -188,57 +188,124 @@ public void testDontSweepFutureConservative() {
...
+    public void testSweepLatestDeletedMultiRowThorough() {
+        createTable(SweepStrategy.THOROUGH);
+        putIntoDefaultColumn("foo", "", 50);
+        put("foo-2", "other", "womp", 60);
+        SweepResults results = completeSweep(75);
+        assertEquals(1, results.getStaleValuesDeleted());
+        assertThat(results.getCellTsPairsExamined()).isGreaterThanOrEqualTo(1);
+
+        assertNull(getFromDefaultColumn("foo", 150));
+        assertEquals(ImmutableSet.of(), getAllTsFromDefaultColumn("foo"));
```
Comment (nziebart) [inline atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/sweep/AbstractSweepTaskRunnerTest.java]: we may want to run another iteration of sweep here, to verify it correctly does nothing if all values are already deleted

Comment (gsheasby) [reply atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/sweep/AbstractSweepTaskRunnerTest.java]: Added on another of the new tests, where we actually have nothing left in the table. There are also a couple of tests (for conservative sweep) where we sweep tables twice.
